qemu/tests/blockdev_stream_subchain.py

```diff
@@ -91,6 +92,12 @@ def _check_backing(self, backing):
                 self.test.fail("Failed to get backing file for %s"
                                % self.snapshot_tag)
             # data->datasn1: check data is datasn1's backing file
+            backing_mask = self.main_vm.check_capability(
+                    Flags.BLOCKJOB_BACKING_MASK_PROTOCOL)
+            raw_format = self.get_image_by_tag(
+                    self.base_tag).image_format == "raw"
+            if backing_mask and raw_format:
+                backing["backing"]["file"] = backing["backing"]
```

Comment:
This changed the backing content, it's better not change it when we doing checks (read-only is better), we can handle this in the same way as the above case

`
backing_dict = backing["backing"] if (backing_mask and raw_format) else backing["backing"]["file"]
if not self._is_same_file(backing_dcit, data_image_opts):
`